### DISCUSSION TITLE
BTCPayServer  - Bitpay API compatbility and format

### DISCUSSION THREAD

**Discussion by boooyu44** [btcpayserver/btcpayserver#2282] [Technical Support] [answered]
Hi,

I want to try to transition from bitpay to btcpayserver.
I'm playing with API using bitpay PHP library with url changed to btcpayserver.
I can create invoices

Problem is with webhooks/instant payment notifications.
The format in which we receive webhooks is NOT bitpay compatible.

```
{
  "deliveryId": "4LEL5EasjPH4D7MrMkqKrp",
  "webhookId": "MvSn3Dzv5Y8zuPjEqCWbc3",
  "orignalDeliveryId": "4LEL5EasjPH4D7MrMkqKrp",
  "isRedelivery": false,
  "type": "InvoiceCreated",
  "timestamp": 1613207436,
  "storeId": "6kUCWkAbgwtBo2bo9SS8HJWyCnGNiVHLoSW2kjcqutw6",
  "invoiceId": "RDn2HGNnj7GGam7UPacL7f"
}
```

Compare this to bitpay webhook: https://bitpay.com/docs/invoice-webhooks

Is this normal? I mean btcpayserver is advertised everywhere as bitpay-api compatible, so is there an option to make IPN callbacks also in same format as bitpay's callbacks?

**Comment by Kukks** [comment, marked as answer]
The webhooks are not meant to be bitpay comaptible. Use the notification
Url when creating an invoice to use the bitpay webhooks

**Reply by boooyu44** [reply]
Thank you. I just tested and it works.

So there are have two separate IPNs:
- webhooks
- notifications

Is this mentioned in documentation anywhere?

**Reply by Kukks** [reply]
Yes, notifications are the Bitpay events and webhooks are the GreenField events.

**Reply by bolatovumar** [reply]
Could someone confirm that this distinction is documented somewhere? @britttttk would you know?

**Reply by britttttk** [reply]
@bolatovumar no we have an open issue for documenting webhooks. But regardless we should add this info to the docs. Maybe in addition to [this FAQ](https://docs.btcpayserver.org/FAQ/FAQ-General/#how-can-i-use-the-btcpay-server-api). I will open an issue.

**Reply by boooyu44** [reply]
Well from my perspective, I'm not sure how it is not documented yet.
Notifications about invoice status changes are one of critical pieces in this whole payment processing thing.
You want to create invoice and at least receive information about its changes. That's minimal usage of the server.

**Reply by britttttk** [reply]
@juouy Open issue in docs: https://github.com/btcpayserver/btcpayserver-doc/issues/852

**Reply by bolatovumar** [reply]
I guess it was missed initially but it's good to know there is an issue for this and it's being worked on.
